Automattic/wp-calypso
# Hosting Trial Requiring CC: Add checkout step to hosting trial

Related to https://github.com/Automattic/dotcom-forge/issues/6577

## Proposed Changes

This PR adds checkout step to the hosting trial flow. 

**Note**
Migrations trial is not yet included

* Adds checkout page
* Remove automatic atomic transfer for trial sites.

![image](https://github.com/Automattic/wp-calypso/assets/47489215/381398a3-dfc2-4516-9e7b-c2e73a55214d)


## Testing Instructions


Pre-requisites
Apply diff D145625-code ( this remove the "assignment payment later" option)
Apply diff D145807-code ( set trial expiration, update user attribute, verify email)
Also use store sandbox and test cards that can be found PCYsg-IA-p2 


* go to `/setup/new-hosted-site`
* verify select Free Trial will take you to checkout page and requires credit card to proceed
* Enter test credit card and proceed to checkout
* Verify site is created with the Creator Trial for 7 days

## Pre-merge Checklist
